File: Data Structures & Algorithms/longest-palindromic-substring/submission-2.js
class Solution {
  /**
   * @param {string} s
   * @return {string}
   */
  longestPalindrome(s) {

    function expandAroundCenter(left, right) {
      while (left >= 0 && right < s.length) {
        if (s[left] !== s[right])
          break;
        left--;
        right++;
      }
      return [left + 1, right - left - 1];
    }
    let ans = "";
    for (let i = 0; i < s.length; i++) {
      let [oddStart, oddLength] = expandAroundCenter(i, i);
      let [evenStart, evenLength] = expandAroundCenter(i, i + 1);
      if (evenLength > ans.length)
        ans = s.substring(evenStart, evenLength + evenStart);
      if (oddLength > ans.length)
        ans = s.substring(oddStart, oddLength + oddStart);

    }
    return ans;

  }
}

File: Data Structures & Algorithms/three-integer-sum/submission-2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()
        prev = None
        for index in range(len(nums)):
            if prev == nums[index]:
                continue
            prev = nums[index]
            left, right = index + 1, len(nums) - 1
            while left < right:
                if nums[index] + nums[left] + nums[right] == 0:
                    ans.append([nums[index], nums[left], nums[right]])
                    right -= 1
                elif nums[index] + nums[left] + nums[right] > 0:
                    right -= 1
                else:
                    left += 1
        return ans

